### PR TITLE
Form file field token as url

### PR DESCRIPTION
**Pull request by kuzmany**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Added two new enhacement to **file field type**:
- set download link to token (for example Send form results action)
- allow match any contact field with download file url

Added new service FieldValueTransformer and unit tests.

#### Steps to test this PR:
1.  Create form with email and file field, add contact field matching

image

2.  Create Send form results action

image

3. Test form, upload file and send results to your email

4. See if file tokens was replaced by download url of file

5. See If contact match field was replaced by download url of uploaded file

Email before:

image

After:

image

### CODE REVIEW

**Comment by kuzmany** [comment]
Label: Enhancement

**Comment by kuzmany** [comment]
Label: WIP

**Comment by kuzmany** [comment]
Label: Ready to test

File: app/bundles/FormBundle/Event/Service/FieldTokenTransformer.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: zdeno
+ * Date: 19. 5. 2018
+ * Time: 14:59.
```
Comment (escopecz) [inline app/bundles/FormBundle/Event/Service/FieldTokenTransformer.php]: Update to Mautic headers please

**Comment by escopecz** [comment]
@kuzmany you'll have to mock that `UploadFieldValidator::processFileValidation()` will return a mock `UploadedFile` object. Do it on this mock object: https://github.com/mautic/mautic/blob/faf191c462aae6b17f41fb80a9f8ecb5d4188768/app/bundles/FormBundle/Tests/FormTestAbstract.php#L315

**Comment by kuzmany** [comment]
@escopecz 
Now should unit test fixed and run as expect

image

**Comment by kuzmany** [comment]
Label: Enhancement

**Comment by kuzmany** [comment]
Label: WIP

**Comment by FerRubioMorales** [comment]
Thanks for this! i think we also comment for thank you, not only for request new features or solve issues... Awesome work.

**Comment by alanhartless** [comment]
This needs a migration right?

**Comment by npracht** [comment]
@escopecz could you review your review please? :)

**Comment by clementdes** [comment]
Thank you for this ;) 
Just when I need it!

**Pull request merged**